File: farm_wise_frontend/lib/models/product.dart
class Product {
  String productId, userId, productName, quantityUnit, productDesc, location;
  DateTime producedDate, availableFrom;
  num basePrice, quantityPerLot, minNoLot, ratings;
  bool availStatus;
  List? productImages = [], reviews = [], ratedUser = [];

  Product({
    required this.productId,
    required this.userId,
    required this.productName,
    required this.quantityUnit,
    required this.productDesc,
    required this.producedDate,
    required this.availableFrom,
    required this.location,
    required this.basePrice,
    required this.quantityPerLot,
    required this.minNoLot,
    required this.ratings,
    required this.availStatus,
    required this.productImages,
    required this.reviews,
    required this.ratedUser,
  });
}

File: farm_wise_backend/model/product.js
const mongoose = require("mongoose");

const ProductNewSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
    },
    productName: {
      type: String,
      required: true,
    },
    basePrice: {
      type: Number,
      required: true,
    },
    quantityPerLot: {
      type: Number,
      required: true,
    },
    quantityUnit: {
      type: String,
      required: true,
    },
    minNoLot: {
      type: Number,
      required: true,
    },
    productDesc: {
      type: String,
      required: true,
    },
    productImages: {
      type: Array,
      required: true,
    },
    producedDate: {
      type: Date,
      required: true,
    },
    availableFrom: {
      type: Date,
      required: true,
    },
    reviews: {
      type: Array,
    },
    ratings: {
      type: Number,
    },
    ratedUser: {
      type: Array,
    },
    location: {
      type: String,
      required: true,
    },
    availStatus: {
      type: Boolean,
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

const Product = mongoose.model("t_products", ProductNewSchema);

module.exports = Product;
